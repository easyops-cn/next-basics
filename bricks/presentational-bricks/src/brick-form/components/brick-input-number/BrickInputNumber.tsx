import React from "react";
import { InputNumber, Spin } from "antd";
import { WrapperFormItem } from "../wrapper-form-item/WrapperFormItem";
import { handleChange } from "../../processor";
import { InputNumberProps } from "antd/lib/input-number";

export interface LegacyBrickInputNumberProps {
  configProps?: InputNumberProps;
  onChange: Function;
  value?: number;
}

export function LegacyBrickInputNumber(
  props: LegacyBrickInputNumberProps
): React.ReactElement {
  const { configProps, onChange, value } = props;

  return (
    <InputNumber
      {...configProps}
      value={value}
      onChange={e => handleChange(e, onChange)}
    />
  );
}

export const BrickInputNumber = WrapperFormItem(LegacyBrickInputNumber);
