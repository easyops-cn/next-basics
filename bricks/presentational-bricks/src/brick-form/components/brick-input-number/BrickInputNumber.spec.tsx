import React from "react";
import { shallow } from "enzyme";
import { LegacyBrickInputNumber } from "./BrickInputNumber";

describe("BrickInputNumber", () => {
  it("should work", () => {
    const mockFn = jest.fn();
    const wrapper = shallow(<LegacyBrickInputNumber onChange={mockFn} />);
    expect(wrapper).toMatchSnapshot();

    wrapper.simulate("change", { target: { value: 4 } });

    expect(mockFn).toHaveBeenCalled();
  });
});
