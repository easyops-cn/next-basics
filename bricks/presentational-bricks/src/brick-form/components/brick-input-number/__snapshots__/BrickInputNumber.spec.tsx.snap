// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrickInputNumber should work 1`] = `
<ForwardRef
  onChange={[Function]}
  step={1}
/>
`;
